includes/Generator/Plugins/SchemaOrg.php

```diff
@@ -178,10 +186,29 @@ private function getAuthorMetadata() {
 			// Empty tags will be ignored
 			$sitename = '';
 		}
-
+		
+		global $wgServer;
+		
+				try {
+			$logo = MediaWikiServices::getInstance()->getMainConfig()->get( 'Logo' );
+			$logo = wfExpandUrl( $logo );
+			$data['url'] = $logo;
+		} catch ( Exception $e ) {
+			// Uh oh either there was a ConfigException or there was an error expanding the URL.
+			// We'll bail out.
+			$data = [];
+		}
+		
+		$logoLine = [
+			'@type' => 'ImageObject',
+			'url' => $logo,
+			'caption' => $sitename
+		];
```

Comment:
Maybe extract this into a separate method which can be called by ```getAuthorMetadata()``` to set the organization logo and by ```getImageMetadata()``` if no image is set?

Reply:
This is a bit of a tricky one, as [$wgLogo](https://www.mediawiki.org/wiki/Manual:$wgLogo) won't actually ever return a blank value if not set; you'd have to outright set `$wgLogo = ""` and even that doesn't really cause an issue in this case. Alternately, you could adopt the admittedly non-standard [Clearbit og:logo tag](https://clearbit.com/logo) and at least allow user control.

Also, extracting it means you wind up extracting out $sitename in order to fill both caption and the main org name, and then it just gets a little odd to have both like that.

Reply:
The idea was to use the sites logo as a fallback in ```getImageMetadata()``` if the user did not supply an image through the extension. So my idea was to extract the part with ```MediaWikiServices::getInstance()->getMainConfig()->get( 'Logo' ); [...]``` into a separate method that ```getImageMetadata()``` could call if ```$this->metadata['image']``` is not set.